sda-commons-shared-otel-instrumentation/README.md

```diff
@@ -0,0 +1,82 @@
+
+# Docs
+An extensive documentation can be found in [OpenTelemetry Java documentation](https://opentelemetry.io/docs/instrumentation/java/). 
+
+# Usage
+
+This bundle requires the [openTelemetry agent](https://mvnrepository.com/artifact/io.opentelemetry.javaagent/opentelemetry-javaagent) to be added as a runtime dependency to the application.
+Sda-commons already provides an extension of the agent [sda-commons-shared-otel-agent](../sda-commons-shared-otel-agent) that can be used.
+
+```groovy
+  api project(':sda-commons-shared-otel-instrumentation')
+  runtimeOnly project(':sda-commons-shared-otel-agent')
+```
+
+Then the bundle is ready to be added to the application.
+
+```java
+public class MyApp extends Application<Configuration> {
+  @Override
+  public void run(Configuration configuration, Environment environment) {}
+
+  @Override
+  public void initialize(Bootstrap<Configuration> bootstrap) {
+    // ... other bundles  
+    bootstrap.addBundle(InstrumentationBundle.builder().build());
+  }
+}
+```
+
+If the application is already using the [starter bundle](../sda-commons-server-starter), no changes are needed.
+The agent is loaded and ready to be configured with environment variables.
+
+> **_NOTE:_** If the application is using the starter bundle, which is already using the [sda-commons-shared-otel-agent](../sda-commons-shared-otel-agent), and it is required to use another vendor extension, the `sda-commons-shared-otel-agent` module must be excluded from the starter bundle.
+
+# Agent configuration
+
+The agent is highly configurable! Many aspects of the agent's behavior can be configured for your needs, such as exporter choice, exporter config (like where data is sent), trace context propagation headers...
+The configuration can be done either with environment variables or system properties. Please note that system properties will have a higher priority over environment variables.
+
+A full list of the configurable properties can be found in the [autoconfigure module](https://github.com/open-telemetry/opentelemetry-java/tree/main/sdk-extensions/autoconfigure#exporters).
+
+It is recommended to use same configuration across all services in the same domain, to ensure the right context propagation, and export traces to the same monitoring backend.
+
+## Service Name
+
+The service name is used to identify the source (the sender) of the received telemetry data in the monitoring backend. Therefore it's important that every service has a unique name.
+To provide the application with a custom name for each deployment, the `OTEL_SERVICE_NAME` environment variable must be used.
+
+## Basic configuration
+
+By default, the agent exports traces in the otlp format, to an [openTelemetry collector](https://opentelemetry.io/docs/collector/). The most basic configuration would be to provide the service name and a collector endpoint.
+- `OTEL_SERVICE_NAME=my-service`
```

Comment:
The JaegerBundle used the application name as default service name. What happens here if you don't specify OTEL_SERVICE_NAME?

Reply:
the default will be `unknown_service:java`. not sure if this can be changed to keep the current behaviour, since it is set in a separate process.